test_rails/generators/flipper/setup_generator_test.rb

```diff
@@ -0,0 +1,64 @@
+require "helper"
+require "generators/flipper/setup_generator"
+
+class SetupGeneratorTest < Rails::Generators::TestCase
+  tests Flipper::Generators::SetupGenerator
+  ROOT = File.expand_path("../../../tmp/generators", __dir__)
+  destination ROOT
+  setup :prepare_destination
+
+  test "invokes flipper:active_record generator if ActiveRecord adapter defined" do
+    begin
+      load 'flipper/adapters/active_record.rb'
+      run_generator
+      assert_migration "db/migrate/create_flipper_tables.rb"
+    ensure
+      Flipper::Adapters.send(:remove_const, :ActiveRecord)
+    end
+  end
+
+  test "does not invoke flipper:active_record generator if ActiveRecord adapter not defined" do
+    # Ensure adapter not defined
+    Flipper::Adapters.send(:remove_const, :ActiveRecord) rescue nil
+
+    run_generator
+    assert_no_migration "db/migrate/create_flipper_tables.rb"
+  end
+
+  %w(.env.development .env.local .env).each do |file|
+    test "configures Flipper Cloud token in #{file} if it exists" do
+      File.write("#{ROOT}/#{file}", "")
+      run_generator %w(--token abc123)
+      assert_file file, /^FLIPPER_CLOUD_TOKEN=abc123$/m
+    end
+  end
+
+  test "configures Flipper Cloud token in .env.development before .env" do
+    File.write("#{ROOT}/.env.development", "")
```

Comment:
Is there some test magic that removes these files (and credentials) in setup or teardown or do they all need to be rm'd?

Reply:
You know this. https://github.com/flippercloud/flipper/pull/804/files/87d8a2954a1a3464178d4741f4755bf577ed3f0c..4748609b66601a6e94d96dad4ae2e958084a9039#diff-20ac1d85392c5779ec3830ff97c1ed050bc0fb9897b576cb795245320c4853eeL6-L8